core/src/main/java/org/apache/calcite/rel/RelCollations.java

```diff
@@ -227,6 +247,21 @@ public static boolean containsOrderless(List<RelCollation> collations,
     return false;
   }
 
+  /**
+   * Returns whether one of a list of collations is contained by the given list of keys
+   * regardless the order.
+   */
+  public static boolean keysContainCollationsOrderless(
+      List<Integer> keys,  List<RelCollation> collations) {
+    final List<Integer> distinctKeys = Util.distinctList(keys);
```

Comment:
Given that `containsOrderless` already does a `Util.distinctList(keys)`, we could skip it here.
The same applies to `collationsContainKeysOrderless`